src/shader/shaderMaterialSampling.js

```diff
@@ -272,6 +276,27 @@ void clearcoatColor( inout vec3 color, vec3 wo, vec3 wi, SurfaceRec surf ) {
 
 }
 
+// sheen
+vec3 sheenColor( vec3 wo, vec3 wi, SurfaceRec surf ) {
+
+	vec3 halfVector = getHalfVector( wo, wi );
+
+	float cosThetaO = saturateCos( wo.z );
+	float cosThetaI = saturateCos( wi.z );
+	float cosThetaH = halfVector.z;
+
+	float D = velvetD( cosThetaH, surf.sheenRoughness );
+	float G = velvetG( cosThetaO, cosThetaI, surf.sheenRoughness );
+
+	vec3 color = surf.sheenColor;
+	color *= D * G / ( 4.0 * abs( cosThetaO * cosThetaI ) );
+	color *= wi.z;
+
+	return color;
+
+}
```

Comment:
Is this function from a paper or anything? Is there a reference comment we could add?

Reply:
It is the `sheenColor * FS * GS * DS / (4 * abs(dot(N, L)) * abs(dot(N, V)))` from KHR_materials_sheen with the suggestion `The Fresnel term may be omitted, i.e., F = 1.` applied.

You can also check `sheen_layer` function in `sheen.glsl` in the `dspbr-pt` project which uses a similar formulation, just split out differently.